src/In.ProjectEKA.HipService/OpenMrs/DiscoveryDataSource.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Web;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using In.ProjectEKA.HipLibrary.Patient.Model;
+using Patient = Hl7.Fhir.Model.Patient;
+
+namespace In.ProjectEKA.HipService.OpenMrs
+{
+    public class DiscoveryDataSource : IPatientDal
+    {
+        private readonly IOpenMrsClient openMrsClient;
+
+        public DiscoveryDataSource(IOpenMrsClient openMrsClient)
+        {
+            this.openMrsClient = openMrsClient;
+        }
+
+        public async Task<List<Patient>> LoadPatientsAsync(string name, AdministrativeGender? gender, string yearOfBirth)
+        {
+            var path = DiscoveryPathConstants.OnPatientPath;
+            var query = HttpUtility.ParseQueryString(string.Empty);
+            if (!string.IsNullOrEmpty(name)) {
+                query["name"]=name;
+            }
+            if (gender != null) {
+                query["gender"]=gender.ToString().ToLower();
```

Comment:
Codacy found an issue: [Define the locale to be used in this string operation.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5933330)